devel/cutter-r2/Portfile

```diff
@@ -0,0 +1,80 @@
+                    qtsvg
+                    
+compiler.cxx_standard 2011
+configure.args-append \
+                    src/Cutter.pro \
+                    INCLUDEPATH+=${prefix}/include/libr
+
+destroot {
+    copy ${worksrcpath}/Cutter.app ${destroot}${applications_dir}
+}
+
+proc python-depends {python_branch} {
+        global frameworks_dir
+        set ::python_version [string map {. ""} ${python_branch}]
+        set ::python_framework ${frameworks_dir}/Python.framework
+        global python_framework python_version
+
+        depends_lib-append  port:python${python_version} \
+                            port:py${python_version}-pyside2
+
+        configure.args-append \
+                            CONFIG+=CUTTER_ENABLE_PYTHON \
+                            CONFIG+=CUTTER_ENABLE_PYTHON_BINDINGS \
+                            INCLUDEPATH+=${python_framework}/Versions/${python_branch}/include/python${python_branch}/ \
+                            LIBS+="-L${python_framework}/Versions/${python_branch}/lib -lpython${python_branch}" \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/shiboken2/ \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/PySide2/ \
+                            PYTHON_FRAMEWORK_DIR=${python_framework} \
+                            SHIBOKEN_EXECUTABLE=${python_framework}/Versions/${python_branch}/bin/shiboken2 \
+                            SHIBOKEN_INCLUDEDIR=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/shiboken2_generator/include/ \
+                            SHIBOKEN_LIBRARY=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/shiboken2/libshiboken2.cpython-*-darwin.*.dylib \
+                            PYSIDE_INCLUDEDIR=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/PySide2/include/ \
+                            PYSIDE_LIBRARY=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/PySide2/libpyside2.cpython-*-darwin.*.dylib \
+                            PYSIDE_TYPESYSTEMS=${python_framework}/Versions/${python_branch}/lib/python${python_branch}/site-packages/PySide2/typesystems
+
+        pre-destroot {
+            move ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter.bin
+            copy ${filespath}/Cutter ${worksrcpath}/Cutter.app/Contents/MacOS/
+            reinplace "s|@PYFRAMEWORK@|${python_framework}|g" ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter
+            reinplace "s|@PYVER@|${python_branch}|g" ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter
+        }
+}
+
+variant python38 description {Enable Python support and bindings using Python 3.8} {
+    set ::python_branch 3.8
+    python-depends ${::python_branch}
+}
```

Comment:
I guess that works... introducing a proc complicates things though. My personal preference would be for something more like:

```
if {![variant_isset python37] && ![variant_isset python38]} {
    default_variants +python38
}

variant python37 conflicts python38 description {Enable Python support and bindings using Python 3.7} {}
if {[variant_isset python37]} {set python_branch 3.7}

variant python38 conflicts python37 description {Enable Python support and bindings using Python 3.8} {}
if {[variant_isset python38]} {set python_branch 3.8}

if {[exists python_branch]} {
    # now you can write all your other code here using ${python_branch} where needed
}
```

(Just as an example for how multiple variants would be handled; you don't have to add a python37 variant.)

Some other ports use complicated code to programmatically generate a series of python variants. It seems to be popular but I find that harder to understand.

Reply:
I've used this method in some other ports (without the `::` workaround) and i'm kinda used to it. I know there's a foreach loop way to generate variants, but yeah it's complicated.

Reply:
I must admit I'm not familiar with what `::` is doing here.

Reply:
`::` refers to the global namespace so the variable is usable in `pre-destroot`, according to what I found.
https://stackoverflow.com/questions/10179334/how-to-declare-a-variable-globally-which-is-used-only-in-proc

Reply:
It'll be all yours forever, as nobody else (except maybe marcus) will have a clue what it means, or how to fix it when it breaks, I think...

Reply:
Well... I'll add some comments.